### DISCUSSION TITLE
Obtaining tfplan output

### DISCUSSION THREAD

**Discussion by hulquest** [flux-iac/tofu-controller#554] [Q&A] [answered]
I'm not sure if this is a bug or know-how problem on my side so I'll start with a question.

I can decode and unzip tfstate Secrets without issue.  However, using the same technique I can't decode tfplan Secrets into readable text.

`k get secrets tfstate-default-curly-voice-mcjc6s -o jsonpath={.data.tfstate} |base64 -d  |gzip -d` --> good
`k get secrets tfplan-default-curly-voice-mcjc6s -o jsonpath={.data.tfplan} |base64 -d  |gzip -d` --> data still looks compressed.  Heres a sample of the output of tfplan data:
```
                                        h8dAZkvق"p+nTGkB:7%>`>
                                                              ƈ$=wY^>vb
                                                                       ߉e|(;Qj[3<PNN)dק\9=qyK>Epuu|WּZP52
f
hTXl('?ayprۥ
            `-[$!i])ʹ#\z
                        FQFiQ1  (8LnWȅa0J0IF%8]rQQp7Α
1$      ]                                            =]p7LQۛqwk}"
  )p;NucgtOѰ%:W@1ǝn10`
                      A܎"`lB% 0N
l7B`=F  đ)qPCtIn01Fi
```
The code looks fine to me but I don't have a dev env setup to step through with the debugger.  

I wanted to ask the community if there is something I'm missing before logging an issue.

**Comment by hulquest** [comment, marked as answer]
Sometimes you have to ask a question to have a breakthrough.  I figured out what was happening after looking at [a test for tfplan](https://github.com/weaveworks/tf-controller/blob/532e18609d894e2c09c857d7ade60906a2281ff1/tfctl/show_plan_test.go) it occurred to me the output still looks like compressed data...because it is.  The solution is to decompress the data again like:
`k get secrets tfplan-default-curly-voice-mcjc6s -o json |jq -r .data.tfplan |base64 -d  |gzip -d |gzip -d`

I'll close this discussion and open an issue.

**Comment by shencan** [comment]
i get error `gzip: unknown compression format`
`k get secrets tfplan-default-starrocks-flink -n flux-system -o json |jq -r .data.tfplan|base64 -d|gzip -
d|gzip -d`

**Comment by chanwit** [comment]
We now support writing a human readable plan as plain text in a ConfigMap, so you don't have to extract it yourself anymore.
